pkg/infrastructure/vsphere/clusterapi/folder.go

```diff
@@ -11,16 +11,20 @@ import (
 )
 
 func createFolder(ctx context.Context, fullpath string, session *session.Session) (*object.Folder, error) {
+	var notFoundError *find.NotFoundError
 	dir := path.Dir(fullpath)
 	base := path.Base(fullpath)
 	finder := session.Finder
 
 	folder, err := finder.Folder(ctx, fullpath)
+	if err != nil && !errors.As(err, &notFoundError) {
```

Comment:
If you just need to check for equality:
```suggestion
	if err != nil && !errors.Is(err, notFoundError) {
```

Reply:
`errors.Is(err, find.NotFoundError)` does not work as expected. If I provide `find.NotFoundError` I get a compiler error:

```
Cannot use 'find.NotFoundError' (type NotFoundError struct { kind string path string }) as the type error Type does not implement 'error' as the 'Error' method has a pointer receiver
```

If I try to do something like this

```
	notFoundErrorIs := find.NotFoundError{}
	notFoundIs := errors.Is(err, notFoundErrorIs)
```
returns `false` because the values of the struct are set to empty which doesn't equal in value.

I think this stackoverflow sums up my experience with this:
https://stackoverflow.com/a/76918940

moving back to errors.As()

Reply:
Golang is such a pain.